File: fixtures/scale/swift/large/Abstractions/FamilyDescriptor.swift
// Metadata plus factory methods for one cipher family; every family
// under Families/ provides exactly one conforming type.
//
// FamilyCatalog.all() collects one instance of each descriptor,
// which is how Harness discovers every family to run.
protocol FamilyDescriptor {
    var family: String { get }
    var suite: String { get }
    var kind: CipherKind { get }
    func cipher() -> Cipher
    func vectors() -> [TestVector]
}

File: fixtures/scale/php/large/src/Abstractions/FamilyDescriptor.php
<?php

namespace CipherLab\Abstractions;

// Metadata plus factory methods for one cipher family; every family
// under Families/ provides exactly one implementing class.
interface FamilyDescriptor
{
    public function family(): string;
    public function suite(): string;
    public function cipher(): Cipher;

    /** @return TestVector[] */
    public function vectors(): array;
}
